src/lib/merkle-distributor/index.ts

```diff
@@ -30,11 +30,16 @@ export const createDistribution = (
   giftAmount: BigNumber,
   previousDistribution?: Partial<MerkleDistributorInfo>
 ): MerkleDistributorInfo => {
-  const totalGive = Object.values(gifts).reduce((t, v) => t + v);
-  let balances = Object.keys(gifts).map(address => ({
-    address,
-    earnings: giftAmount.mul(gifts[address]).div(totalGive),
-  }));
+  let balances: Balance[] = [];
```

Comment:
wrapping this whole thing in an if-clause and using `push` isn't necessary--all you need to do is add a default value to `reduce`:
```js
  const totalGive = Object.values(gifts).reduce((t, v) => t + v, 0);
```